mitodl/mitxpro
# feat: replace # with - in Emeritus courserun.courseware_id

### What are the relevant tickets?




N/A

### Description (What does it do?)

Replaces `#` with `-` for the Emeritus `CourseRun.courseware_id`. We recently worked on ingesting the Emeritus API in https://github.com/mitodl/mitxpro/pull/2998. We used the default course tag that we extracted from the API but to keep it consistent and match the regex in https://github.com/mitodl/mitxpro/pull/3023, we decided to update the courseware IDs.

This adds a command as well to fix the courseware IDs on RC.

### How can this be tested?


- If you have already ingested the Emeritus API, you can run `./manage.py update_emeritus_courserun_courseware_id` to update the courseware IDs. Note that `#` is replaced with `-` for the Emeritus CourseRun.courseware_id field.
- If you want to test new data, delete a couple of course runs and run `./manage.py sync_external_course_runs --vendor emeritus`. The new courseware_ids should contain `-` in place of `#`
